diff --git a/node_modules/@oclif/plugin-update/lib/update.js b/node_modules/@oclif/plugin-update/lib/update.js
index b542c89..4b78044 100644
--- a/node_modules/@oclif/plugin-update/lib/update.js
+++ b/node_modules/@oclif/plugin-update/lib/update.js
@@ -333,11 +333,23 @@ class Updater {
         const binPathEnvVar = this.config.scopedEnvVarKey('BINPATH');
         const redirectedEnvVar = this.config.scopedEnvVarKey('REDIRECTED');
         if (windows) {
-            const body = `@echo off
-setlocal enableextensions
-set ${redirectedEnvVar}=1
-set ${binPathEnvVar}=%~dp0${bin}
-"%~dp0..\\${version}\\bin\\${bin}.cmd" %*
+            const body = `@ECHO off
+GOTO start
+:find_dp0
+SET dp0=%~dp0
+EXIT /b
+:start
+SETLOCAL
+CALL :find_dp0
+
+SET "_redirect=1"
+SET "_binpath=%dp0%${bin}"
+SET "_prog=%dp0%..\\${version}\\bin\\${bin}.cmd"
+
+ENDLOCAL & GOTO #_undefined_# 2>NUL || TITLE %COMSPEC% & (
+  SET "${redirectedEnvVar}=%_redirect%"
+  SET "${binPathEnvVar}=%_binpath%"
+) & "%_prog%" %*
 `;
             await fs.outputFile(dst, body);
         }
