diff --git a/node_modules/oclif/lib/commands/pack/deb.js b/node_modules/oclif/lib/commands/pack/deb.js
index 8903363..7a60189 100644
--- a/node_modules/oclif/lib/commands/pack/deb.js
+++ b/node_modules/oclif/lib/commands/pack/deb.js
@@ -64,9 +64,7 @@ class PackDeb extends core_1.Command {
             await qq.write([workspace, 'DEBIAN/control'], scripts.control(buildConfig, (0, upload_util_1.debArch)(arch)));
             await qq.chmod([workspace, 'usr/lib', config.dirname, 'bin', config.bin], 0o755);
             await qq.x(`ln -s "../lib/${config.dirname}/bin/${config.bin}" "${workspace}/usr/bin/${config.bin}"`);
-            await qq.x(`chown -R root "${workspace}"`);
-            await qq.x(`chgrp -R root "${workspace}"`);
-            await qq.x(`dpkg --build "${workspace}" "${qq.join(dist, versionedDebBase)}"`);
+            await qq.x(`dpkg-deb --root-owner-group --build "${workspace}" "${qq.join(dist, versionedDebBase)}"`);
         };
         const arches = _.uniq(buildConfig.targets
             .filter(t => t.platform === 'linux')
diff --git a/node_modules/oclif/lib/commands/promote.js b/node_modules/oclif/lib/commands/promote.js
index d89b9c6..9c3f4a5 100644
--- a/node_modules/oclif/lib/commands/promote.js
+++ b/node_modules/oclif/lib/commands/promote.js
@@ -41,6 +41,7 @@ class Promote extends core_1.Command {
             // eslint-disable-next-line no-await-in-loop
             await aws_1.default.s3.copyObject({
                 Bucket: s3Config.bucket,
+                ACL: s3Config.acl || 'public-read',
                 CopySource: copySource,
                 Key: key,
                 CacheControl: maxAge,
@@ -60,6 +61,7 @@ class Promote extends core_1.Command {
             // eslint-disable-next-line no-await-in-loop
             await aws_1.default.s3.copyObject({
                 Bucket: s3Config.bucket,
+                ACL: s3Config.acl || 'public-read',
                 CopySource: versionedTarGzKey,
                 Key: unversionedTarGzKey,
                 CacheControl: maxAge,
@@ -83,6 +85,7 @@ class Promote extends core_1.Command {
                 // eslint-disable-next-line no-await-in-loop
                 await aws_1.default.s3.copyObject({
                     Bucket: s3Config.bucket,
+                    ACL: s3Config.acl || 'public-read',
                     CopySource: versionedTarXzKey,
                     Key: unversionedTarXzKey,
                     CacheControl: maxAge,
@@ -106,6 +109,7 @@ class Promote extends core_1.Command {
                 // eslint-disable-next-line no-await-in-loop
                 await aws_1.default.s3.copyObject({
                     Bucket: s3Config.bucket,
+                    ACL: s3Config.acl || 'public-read',
                     CopySource: darwinCopySource,
                     Key: darwinKey,
                     CacheControl: maxAge,
@@ -129,6 +133,7 @@ class Promote extends core_1.Command {
                 // eslint-disable-next-line no-await-in-loop
                 await aws_1.default.s3.copyObject({
                     Bucket: s3Config.bucket,
+                    ACL: s3Config.acl || 'public-read',
                     CopySource: winCopySource,
                     Key: winKey,
                     CacheControl: maxAge,
@@ -158,6 +163,7 @@ class Promote extends core_1.Command {
                 // eslint-disable-next-line no-await-in-loop
                 await aws_1.default.s3.copyObject({
                     Bucket: s3Config.bucket,
+                    ACL: s3Config.acl || 'public-read',
                     CopySource: debCopySource,
                     Key: debKey,
                     CacheControl: maxAge,
diff --git a/node_modules/oclif/lib/version-indexes.js b/node_modules/oclif/lib/version-indexes.js
index 2410048..ca12848 100644
--- a/node_modules/oclif/lib/version-indexes.js
+++ b/node_modules/oclif/lib/version-indexes.js
@@ -64,6 +64,7 @@ const appendToIndex = async (input) => {
         Bucket: s3Config.bucket,
         Key: key,
         CacheControl: maxAge,
+        ACL: s3Config.acl || 'public-read',
     });
     // cleans up local fs
     await fs.remove(jsonFileName);
